File: distros/A/Algorithm-VSM/examples/corpus_with_java_and_cpp/RenderGraphics.cc
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Chapter:     Chapter 17  OO For Graphical User Interfaces, A Tour Of Three Toolkits
//
// Section:     Section 17.20  Drawing Shapes,Text,And Images In Qt
//
// The links to the rest of the code in this book are at
//     
//      http://programming-with-objects.com/pwocode.html
//
// For further information regarding the book, please visit
//
//      http://programming-with-objects.com
//



//RenderGraphics.cc

#include <qwidget.h>
#include <qpainter.h>
#include <qapplication.h>
#include <qpixmap.h>

class RenderGraphicsWidget : public QWidget {                     //(I)
public:
    RenderGraphicsWidget();                                       //(J)
protected:
    void   paintEvent( QPaintEvent * );                           //(K)
};

RenderGraphicsWidget::RenderGraphicsWidget() {                    //(L)
    setCaption( "Render Graphics with Qt" );
    setBackgroundColor( white );
}

void RenderGraphicsWidget::paintEvent( QPaintEvent* ) {           //(M)
    QWMatrix matrix;                                              //(N)
    QPainter painter( this );                                     //(O)

    QBrush b1( Qt::NoBrush );                                    //(P1)
    QBrush b2( Qt::magenta );                                    //(P2)
    QBrush b3( Qt::red, Qt::Dense2Pattern );                     //(P3)
    QBrush b4( Qt::blue, Qt::Dense7Pattern );                    //(P4)
    QBrush b5( Qt::CrossPattern );                               //(P5)

    painter.setPen( Qt::red );                                    //(Q)
    painter.setBrush( b1 );                                       //(R)
    painter.drawRect( 10, 10, 100, 50 );                          //(S)

    matrix.translate( 150, 0 );
    matrix.rotate( (float)3*10 );       
    painter.setWorldMatrix( matrix );
    painter.setBrush( b2 );
    painter.drawRoundRect( 10, 10, 100, 50, 30, 30 );

    matrix.rotate( - (float)3*10 );     
    matrix.translate( - 150, -30 );
    matrix.shear( 0.8, 0.2 );           
    painter.setWorldMatrix( matrix );
    painter.setBrush( b3 );
    painter.drawRect( 250, 0, 100, 50 );

    matrix.reset();                                               //(T)
    //  matrix.setMatrix( 1.0, 0.0, 0.0, 1.0, 0.0, 0.0 );
    painter.setWorldMatrix( matrix );
    painter.setBrush( b4 );
    painter.drawRect( 10, 200, 100, 50 );

    painter.setBrush( b5 );
    painter.drawEllipse( 130, 200, 100, 50 );

    painter.setPen( Qt::NoPen );
    QPixmap pix( "allthatjazz.xpm" );
    pix.resize( 100, 50 );
    painter.drawPixmap( 260, 200, pix );

    int y = 300;
    painter.setWorldMatrix( matrix );
    QFont font( "Times", 18 );
    painter.setFont( font );
    QFontMetrics fm = painter.fontMetrics();
    y += fm.ascent();
    painter.drawText( 70, y, 
                      "Graphics rendered using QPainter methods" );
}

int main( int argc, char **argv )
{
    QApplication app( argc, argv );
    RenderGraphicsWidget drawdemo;
    drawdemo.setGeometry( 200, 200, 450, 400 );
    app.setMainWidget( &drawdemo );
    drawdemo.show();
    return app.exec();
}

File: distros/A/Algorithm-VSM/examples/corpus_with_java_and_cpp/RenderGraphics.java
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Chapter:     Chapter 17  OO For Graphical User Interfaces, A Tour Of Three Toolkits
//
// Section:     Section 17.19  Drawing Shapes, Text, And Images In AWT/Swing
//
// The links to the rest of the code in this book are at
//     
//      http://programming-with-objects.com/pwocode.html
//
// For further information regarding the book, please visit
//
//      http://programming-with-objects.com
//



//RenderGraphics.java

import java.awt.*;            // for Color, GridLayout, Graphics, etc.
import java.awt.event.*;      // for WindowAdapter
import javax.swing.*;

public class RenderGraphics {
    static final int maxCharHeight = 15;
    static final Color bg = Color.lightGray;
    static final Color fg = Color.black;
    static int width;          // for width of a shape panel
    static int height;         // for height of a shape panel
    static int rectWidth;      // width of shape's bounding rect
    static int rectHeight;     // height of shape's bounding rect

    public static void main( String[] args ) {
        JFrame f = new JFrame( "Draw Shape Samples" );
        f.addWindowListener(new WindowAdapter() {
            public void windowClosing(WindowEvent e) {
                System.exit(0);
            }
        });

        f.setBackground(bg);                                      //(D) 
        f.setForeground(fg);                                      //(E)

        Container contentPane = f.getContentPane();
        contentPane.setLayout( new GridLayout( 0, 3 ) );

        //polyline:
        contentPane.add( new JPanel() {
            public void paintComponent(Graphics g) {
                super.paintComponent(g);      //clears the background
                width = getWidth();
                height = getHeight();
                int stringY = height - 10;
                rectWidth = width - 20;
                rectHeight = stringY - maxCharHeight - 10;
                int x2Points[] = 
                      {10, 10+rectWidth, 10, 10+rectWidth};
                int y2Points[] = 
                      {10, 10+rectHeight, 10+rectHeight, 10};
                g.drawPolyline(x2Points, 
                      y2Points, x2Points.length);                 //(F)
                g.drawString("drawPolyline", 10, stringY);        //(G)
            }
            });

        //rounded rectangle:
        contentPane.add( new JPanel() {
            public void paintComponent(Graphics g) {
                super.paintComponent(g);  
                width = getWidth();
                height = getHeight();
                int stringY = height - 10;
                rectWidth = width - 20;
                rectHeight = stringY - maxCharHeight - 10;
                g.drawRoundRect(10, 
                      10, rectWidth, rectHeight, 10, 10);         //(H)
                g.drawString("drawRect", 10, stringY);        
            }
            });

        //filled oval:
        contentPane.add( new JPanel() {
            public void paintComponent(Graphics g) {
                super.paintComponent(g);  
                width = getWidth();
                height = getHeight();
                int stringY = height - 10;
                rectWidth = width - 80;
                rectHeight = stringY - maxCharHeight - 10;
                g.fillOval(40, 10, rectWidth, rectHeight);        //(I)
                g.drawString("drawOval", 10, stringY);
              }
            });

        f.setSize(new Dimension(550, 200));
        f.setVisible(true);
    }
}